spec/index.bs

```diff
@@ -5414,6 +5414,48 @@ dictionary GPUImageCopyTexture {
 
 Issue(gpuweb/gpuweb#69): Define the copies with {{GPUTextureDimension/1d}} and {{GPUTextureDimension/3d}} textures.
 
+### <dfn dictionary>GPUImageCopyTextureTagged</dfn> ### {#gpu-image-copy-texture-tagged}
+
+WebGPU textures hold raw numeric data, and are not tagged with semantic metadata describing colors.
+However, {{GPUQueue/copyExternalImageToTexture()}} copies from sources that describe colors.
+
+A {{GPUImageCopyTextureTagged}} is a {{GPUImageCopyTexture}} which is additionally tagged with
+color space/encoding and alpha-premultiplication metadata, so that semantic color data may be
+preserved during copies.
+This metadata only affects the semantics of the {{GPUQueue/copyExternalImageToTexture()}} call.
+
+<script type=idl>
+dictionary GPUImageCopyTextureTagged : GPUImageCopyTexture {
+    GPUPredefinedColorSpace colorSpace = "srgb";
+    boolean premultipliedAlpha = false;
```

Comment:
Open question: Should this be required? The default of `false` incurs an extra cost for sources which are already premultiplied (like 2d canvases).

Alternatively, should this be tristate? "If undefined, no premultiplication or unpremultiplication is applied." (I believe some image sources have implementation-defined premul-state, in which cases this isn't portable so we would have to carve out exceptions to make it portable.)

Reply:
Since it's a copy, why does it care about alpha premultiplication? I thought it only matters for compositing.

Reply:
Copy sources may be either premultiplied or not, see https://github.com/gpuweb/gpuweb/issues/1762#issuecomment-846203139

So just like with color spaces, the source is "tagged" with whether it's premultiplied or unpremultiplied. When we do an upload we want the user to be able to define which result they get. We have to do one of:
- Specify result is always the same (e.g. always unpremultiplied)
- Specify different results for different sources
- Make it configurable

Reply:
And the argument for doing it during the copy (3) is that it's more efficient than whatever user would have to do if they were to change the property?
Would it be reasonable to expect the user to set this on the source image to match their destination? I.e. if I need a premultiplied alpha, I'm going to have a canvas configured this way, avoid any conversion (or configuration in the copy).

Reply:
> And the argument for doing it during the copy (3) is that it's more efficient than whatever user would have to do if they were to change the property?

If I understand the question, the alternative is that the user does the conversion using createImageBitmap. Depending on their source, this could mean an extra copy. (OTOH if they're coming from HTMLImageElement, they could have done the conversion during the createImageBitmap that was already necessary, so it's roughly the same.)

> Would it be reasonable to expect the user to set this on the source image to match their destination? I.e. if I need a premultiplied alpha, I'm going to have a canvas configured this way, avoid any conversion (or configuration in the copy).

I don't think we can design reasonable semantics that say there is never a conversion in copyExternalImageToTexture, since internal formats of some sources (like ImageBitmap) can be implementation defined (IIUC). However if we wanted we could say:
- If the source is a 2d canvas, webgl/2 canvas with `premultipliedAlpha: true`, or ImageBitmap created with `premultiplyAlpha: "premultiply"`, then `GPUImageCopyTextureTagged.premultipliedAlpha` must be true
    - Problematic because I think applications are very likely to want to upload their 2d-canvas-rendered text as unpremultiplied textures.
- If the source is a webgl/2 canvas with `premultipliedAlpha: false` or an ImageBitmap created with `premultiplyAlpha: "none"`, then `GPUImageCopyTextureTagged.premultipliedAlpha` must be false
- If the source is an ImageBitmap created with `premultiplyAlpha: "default"`, then we may do a conversion (or alternatively we require `GPUImageCopyTextureTagged.premultipliedAlpha` to be true or something).

Meanwhile we still may need conversions due to `colorSpace`, so we would still need color space conversions, or apply equivalent validation rules to `colorSpace`.